java/sage/BigBrother.java

```diff
@@ -189,6 +189,7 @@ public static void clearWatched(Airing a)
     triggerWatchedChangeEvent(a);
   }
 
+  public static boolean alwaysUseAiringWatched = Sage.getBoolean("wizard/always_use_airing_for_watched_status", false);
```

Comment:
No problem; I'll change the variable type(s).  Do you want me to also change the advRedCheck type too?  I'm not sure I follow what you're suggesting.  The code starts with the airing duration then selects the show duration instead if it's within 20 minutes of the airing duration or greater.  Isn't he airing duration usually longer than the show because it includes commercials?  The intent of the new boolean is to force the use of airing duration for consideration for 'watched' regardless of show length.  The opposite of "always airing" would be something like 'dontUseShowTimeForWatched'.  In your suggestion does 'running time' mean the airing time or show time?  My understanding of airing -vs- show may be off.

I would also want to change the property to match.  i.e. something like "wizard/never_use_show_duration_for_watched_status" to fully describe it.

Reply:
Don't change advRedCheck because it's actually used outside of the class (Seeker actually updates it, very dirty code).

Some background...the duration in Show is only ever populated when there's a 'running time' which is only for movies. This is the length of the movie itself. So that could be 106 minutes let's say. But on HBO, maybe they'll put that into a 120 minute block...but we want to mark it watched when it's only about 102 minutes in (because of the credits). So we use the Show duration (i.e. running time) to change that 120 to be 106...and then we have the 10 minute watch ignore time to cover that difference between 102 and 106 (sometimes credits are over 5 minutes, which is why we have a 10 minute difference for movies vs. 5 minutes for TV shows).

There's also that ugly 20 minute hack to realize that it's a movie with commercials...so in that case the running time doesn't help us at all and we ignore that.

So for me...calling it 'dontUseRunningTimeForWatched' makes perfect sense. :) What may make sense to more people is calling it 'dontUseShowDurationForWatched'...then the whole running time thing doesn't matter as much.

Reply:
Reviewing graywolf's post, the UI appears to refer to them as 'recording' and 'run' time so your name makes sense in that context.  He also seems to have a TV show with 'run' time so maybe the SD EPG provides that now?  Anyway I thnk I'll go with your suggestion and name the property accordingly.